zrml/orderbook/src/mock.rs

```diff
@@ -38,6 +43,41 @@ use zeitgeist_primitives::{
 pub const ALICE: AccountIdTest = 0;
 pub const BOB: AccountIdTest = 1;
 
+pub const MARKET_CREATOR: AccountIdTest = 42;
+
+pub const INITIAL_BALANCE: Balance = 100 * BASE;
+
+parameter_types! {
+    pub const FeeAccount: AccountIdTest = MARKET_CREATOR;
+}
+
+pub struct ExternalFees<T, F>(PhantomData<T>, PhantomData<F>);
+
+impl<T: crate::Config, F> DistributeFees for ExternalFees<T, F>
+where
+    F: Get<T::AccountId>,
+{
+    type Asset = AssetOf<T>;
+    type AccountId = T::AccountId;
+    type Balance = BalanceOf<T>;
+    type MarketId = MarketIdOf<T>;
+
+    fn get_fee(_market_id: Self::MarketId, amount: Self::Balance) -> Self::Balance {
+        Perbill::from_rational(1u64, 100u64).mul_floor(amount.saturated_into::<BalanceOf<T>>())
+    }
+
+    fn distribute(
+        _market_id: Self::MarketId,
+        asset: Self::Asset,
+        account: &Self::AccountId,
+        amount: Self::Balance,
+    ) -> Self::Balance {
+        let fees = Self::get_fee(_market_id, amount);
+        let _ = T::AssetManager::transfer(asset, account, &F::get(), fees);
+        fees
```

Comment:
Looking at this now, this implementation is wrong, isn't it? Because if the transfer fails, then the return value should be zero.

```suggestion
        match T::AssetManager::transfer(asset, account, &F::get(), fees) {
            Ok(_) => fees,
            Err(_) => Zero::zero(),
        }
```